plugin/pkg/scheduler/schedulercache/cache_test.go

```diff
@@ -506,6 +505,142 @@ func TestForgetPod(t *testing.T) {
 	}
 }
 
+func TestAddNode(t *testing.T) {
+	// Test data
+	nodeName := "test-node"
+	cpu_1 := resource.MustParse("1000m")
+	mem_100m := resource.MustParse("100m")
+	cpu_half := resource.MustParse("500m")
+	mem_50m := resource.MustParse("50m")
+	resourceFooName := "pod.alpha.kubernetes.io/opaque-int-resource-foo"
+	resourceFoo := resource.MustParse("1")
+
+	cache := newSchedulerCache(time.Second, time.Second, nil)
+
+	test := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nodeName,
+		},
+		Status: v1.NodeStatus{
+			Allocatable: v1.ResourceList{
+				v1.ResourceCPU:                   cpu_1,
+				v1.ResourceMemory:                mem_100m,
+				v1.ResourceName(resourceFooName): resourceFoo,
+			},
+		},
+		Spec: v1.NodeSpec{
+			Taints: []v1.Taint{
+				{
+					Key:    "test-key",
+					Value:  "test-value",
+					Effect: v1.TaintEffectPreferNoSchedule,
+				},
+			},
+		},
+	}
+
+	expected := NewNodeInfo()
+	// Simulate AddNode.
+	expected.node = test
+	expected.allocatableResource = &Resource{
```

Comment:
move it out into some helper func in `cache_test.go`; and build a loop for different cases.